widget/charts/src/components/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,381 @@
+    };
+  }, [tooltipRef]);
+
+  dateScale.range([0, xMax]);
+  valueScale.range([yMax, 0]);
+
+  if (width < 10) {
+    return null;
+  }
+
+  const handleBarClick = (
+    event: React.MouseEvent<SVGRectElement, MouseEvent>,
+    bar: Omit<BarGroupBar<string>, 'key' | 'value'> & {
+      bar: SeriesPoint<BarStackDataType>;
+      key: string;
+    },
+    index: number
+  ) => {
+    if (isMobile) {
+      if (tooltipTimeout) {
+        clearTimeout(tooltipTimeout);
+      }
+      const eventSvgCoords = localPoint(event);
+      const left = bar.x + bar.width / 2;
+      setTimeout(() => {
+        showTooltip({
+          tooltipData: { bar, hoveredIndex: index },
+          tooltipTop: eventSvgCoords?.y,
+          tooltipLeft: left,
+        });
+      }, TOOLTIP_DELAY_MS);
+    }
+  };
+
+  const handleMouseLeave = () => {
+    if (!isMobile) {
+      tooltipTimeout = window.setTimeout(() => {
+        hideTooltip();
+      }, TOOLTIP_HIDE_DELAY_MS);
+    }
+  };
+
+  const handleMouseMove = (
+    event: React.MouseEvent<SVGRectElement, MouseEvent>,
+    bar: Omit<BarGroupBar<string>, 'key' | 'value'> & {
+      bar: SeriesPoint<BarStackDataType>;
+      key: string;
+    },
+    index: number
+  ) => {
+    if (!isMobile) {
+      if (tooltipTimeout) {
+        clearTimeout(tooltipTimeout);
+      }
+      /*
+       * TooltipInPortal expects coordinates to be relative to containerRef
+       * localPoint returns coordinates relative to the nearest SVG, which
+       * is what containerRef is set to in this example.
+       */
+      const eventSvgCoords = localPoint(event);
+      const left = bar.x + bar.width / 2 + 40;
+
+      // make sure to pass the index of the hovered bar
+      showTooltip({
+        tooltipData: { bar, hoveredIndex: index },
+        tooltipTop: eventSvgCoords?.y,
+        tooltipLeft: left,
+      });
+    }
+  };
+
+  return (
+    <Container>
+      <svg ref={containerRef} width={width} height={height}>
+        <rect
+          x={0}
+          y={0}
+          width={width}
+          height={height}
+          fill="transparent"
+          rx={14}
+        />
+        <Grid
+          top={margin.top}
+          left={margin.left + 10}
+          xScale={dateScale}
+          yScale={valueScale}
+          width={xMax}
+          height={yMax}
+          stroke="black"
+          strokeOpacity={0.1}
+          numTicksRows={5}
+          numTicksColumns={0}
+          columnLineStyle={{ display: 'none' }}
+        />
+
+        <Group top={margin.top} left={margin.left + 10}>
+          <BarStack
+            data={data}
+            keys={buckets}
+            x={getDate}
+            xScale={dateScale}
+            yScale={valueScale}
+            color={colorScale}>
+            {(barStacks) => {
+              /*
+               * barStacks returns an array of series objects broken down by key.
+               * in this case we've got blockchain name
```

Comment:
This line should be deleted:

`in this case we've got blockchain name`

Reply:
fixed